### PR TITLE
Check for allow combinations setting on variation quantity rules

### LINKED ISSUE #8773: [Min/Max Quantities] Support allow combinations
The `allow combination` is a setting that enables combining different variations purchased to match the quantities rules. When this setting is enabled we should hide the per-variation quantity rules to match `wp-admin` behavior.

> Enable this option to combine the quantities of all purchased variations when checking 'Minimum/Maximum quantity' and 'Group of' rules. **Note:** Cannot be used together with variation-level quantity rules.

### PR DESCRIPTION
**Pull request by atorresveiga**
⚠️ This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2705)

Closes: #8773

### Why
We are adding Min / Max Quantities read-only support to the Woo App. The allow combination is a setting that enables combining different variations purchased to match the quantities rules. When this setting is enabled we should hide the per-variation quantity rules to match wp-admin behavior.

### Description
This PR updates the GetProductVariationQuantityRules behavior to display variations quantity rules only when the combine variations setting is disabled on `wp-admin`.

### Testing instructions

TC1
#### On `wp-admin`
1. Create a variable product with variations.
2. Add quantity rules to one/all of the variations of the variable product
3. Check that the combine variations setting is unchecked on the `wp-admin`

#### On the app
1. Open the products tab
2. Tap on the variable product created
3. Tap on Variations
4. Tap on the variation with the quantities rules
5. Check that the Quantity rules card is shown
6. Tap on the Quantity rules card
7. Check that the Quantity rules information is shown

TC2
#### On `wp-admin`
3. Enabled the combine variations setting 

#### On the app
1. Open the products tab
2. Tap on the variable product created
3. Tap on Variations
4. Tap on the variation with the quantities rules
5. Check that the Quantity rules card is NOT shown

### Images/gif

https://user-images.githubusercontent.com/18119390/230685629-08e249e9-80c4-4ed8-b39d-31be4c91bd06.mp4

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailRepository.kt [hunk 1]
```diff
@@ -70,6 +72,8 @@ class ProductDetailRepository @Inject constructor(
     private var isFetchingTaxClassList = false
     private var remoteProductId: Long = 0L
 
+    private val gson by lazy { Gson() }
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailRepository.kt]: ⛏️ Non blocking: maybe you could inject this, so other classes could re-use the same instance of Gson? I see we use `by lazy { Gson() }` in many places in the app but I can't find a reason for not re-using the single instance.

**Review by wzieba — APPROVED** [review]
Works as described in the test cases, nice work!

---

On the video and during my testing I can see that we display the value of `null` to users. I know it's not the scope of this PR, just a friendly reminder that there is such issue:

Screenshot 2023-04-10 at 18 33 48

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailRepository.kt (hunk 1 above)
Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailRepository.kt]: Fixed here https://github.com/woocommerce/woocommerce-android/pull/8777/commits/262ee9077fae0c446531a6f64ed296d57386cfd4

**Comment by atorresveiga** [comment]
> On the video and during my testing I can see that we display the value of `null` to users. I know it's not the scope of this PR, just a friendly reminder that there is such issue

Thanks for pointing that out @wzieba . As you mentioned, that was out of the scope of this PR, so we tackled it on this other PR: https://github.com/woocommerce/woocommerce-android/pull/8774

**Pull request merged**